src/Http/Cookie/CookieCryptTrait.php

```diff
@@ -0,0 +1,194 @@
+     * Encryption cipher
+     *
+     * @param string
+     */
+    protected $encryptionCipher = 'aes';
+
+    /**
+     * The key for encrypting and decrypting the cookie
+     *
+     * @var string
+     */
+    protected $encryptionKey = '';
+
+    /**
+     * Sets the encryption cipher
+     *
+     * @param string $cipher Cipher
+     * @return $this
+     */
+    public function setEncryptionCipher($cipher)
+    {
+        $this->checkCipher($cipher);
+        $this->encryptionCipher = $cipher;
+
+        return $this;
+    }
+
+    /**
+     * Sets the encryption key
+     *
+     * @param string $key Encryption key
+     * @return $this
+     */
+    public function setEncryptionKey($key)
+    {
+        $this->encryptionKey = $key;
+
+        return $this;
+    }
+
+    /**
+     * Returns the encryption key to be used.
+     *
+     * @return string
+     */
+    public function getEncryptionKey()
+    {
+        if (empty($this->encryptionKey)) {
+            return Security::salt();
+        }
+
+        return $this->encryptionKey;
+    }
+
+    /**
+     * Encrypts $value using public $type method in Security class
+     *
+     * @param string $value Value to encrypt
+     * @param string|bool $encrypt Encryption mode to use. False
+     *   disabled encryption.
+     * @param string|null $key Used as the security salt if specified.
+     * @return string Encoded values
+     */
+    protected function _encrypt($value, $encrypt, $key = null)
+    {
+        if (is_array($value)) {
+            $value = $this->_flatten($value);
+        }
+
+        $this->checkCipher($encrypt);
+        $prefix = 'Q2FrZQ==.';
+        $cipher = null;
+
+        if ($key === null) {
+            $key = $this->getEncryptionKey();
+        }
+        if ($encrypt === 'rijndael') {
+            $cipher = Security::rijndael($value, $key, 'encrypt');
+        }
+        if ($encrypt === 'aes') {
+            $cipher = Security::encrypt($value, $key);
+        }
+
+        return $prefix . base64_encode($cipher);
+    }
+
+    /**
+     * Helper method for validating encryption cipher names.
+     *
+     * @param string $encrypt The cipher name.
+     * @return void
+     * @throws \RuntimeException When an invalid cipher is provided.
+     */
+    protected function checkCipher($encrypt)
+    {
+        if (!in_array($encrypt, $this->_validCiphers)) {
+            $msg = sprintf(
+                'Invalid encryption cipher. Must be one of %s.',
+                implode(', ', $this->_validCiphers)
+            );
+            throw new RuntimeException($msg);
+        }
+    }
+
+    /**
+     * Decrypts $value using public $type method in Security class
+     *
+     * @param string|array $values Values to decrypt
+     * @param string|bool $mode Encryption mode
+     * @param string|null $key Used as the security salt if specified.
+     * @return string|array Decrypted values
+     */
+    protected function _decrypt($values, $mode, $key = null)
```

Comment:
Should there be `$encrypt` instead of `$mode`?

Plus same as above (nullable `$key`)